packages/network-controller/tests/fake-provider.ts

```diff
@@ -71,33 +70,23 @@ interface FakeProviderEngineOptions {
 }
 
 /**
- * Implements just enough of the block tracker interface to pass the tests but
- * nothing more.
+ * An implementation of the provider that NetworkController exposes, which is
+ * actually an instance of SafeEventEmitterProvider (from the
+ * `@metamask/eth-json-rpc-provider` package). Hence it supports the same
+ * interface as SafeEventEmitterProvider, except that fake responses for any RPC
+ * methods that are accessed can be supplied via an API that is more succinct
+ * than using Jest's mocking API.
  */
-class FakeBlockTracker extends EventEmitter {}
-
-/**
- * FakeProviderEngine is an implementation of the provider that
- * NetworkController exposes, which is actually an instance of
- * Web3ProviderEngine (from the `web3-provider-engine` package). Hence it
- * supports the same interface as Web3ProviderEngine, except that fake responses
- * for any RPC methods that are accessed can be supplied via an API that is more
- * succinct than using Jest's mocking API.
- */
-export class FakeProviderEngine extends EventEmitter implements ProviderEngine {
+// NOTE: We shouldn't need to extend from the "real" provider here, but
+// we'd need a `SafeEventEmitterProvider` _interface_ and that doesn't exist (at
+// least not yet).
+export class FakeProvider extends SafeEventEmitterProvider {
   calledStubs: FakeProviderStub[];
 
   #originalStubs: FakeProviderStub[];
 
   #stubs: FakeProviderStub[];
 
-  #isStopped: boolean;
-
-  // NOTE: This is a "private" property defined on ProviderEngine that holds the
-  // block tracker, so we have to provide that in this fake implementation as
-  // well
-  _blockTracker: FakeBlockTracker;
```

Comment:
We don't need this property anymore because the SafeEventEmitterProvider interface doesn't require it.